filament/src/Scene.cpp

```diff
@@ -199,6 +199,13 @@ void FScene::updateUBOs(utils::Range<uint32_t> visibleRenderables, backend::Hand
         mat3f m = mat3f::getTransformForNormals(model.upperLeft());
         m *= mat3f(1.0f / std::sqrt(max(float3{length2(m[0]), length2(m[1]), length2(m[2])})));
 
+        // The shading normal must be flipped for mirror transformations.
+        // Basically we're shading the other side of the polygon and therefore need to negate the
+        // normal, similar to what we already do to support double-sided lighting.
+        if (sceneData.elementAt<REVERSED_WINDING_ORDER>(i)) {
+            m = -m;
```

Comment:
I can't convince myself it is correct to flip T, B and N as this changes the handed-ness -- is this how others are doing it? Wouldn't that mirror the uv? is that what we want?

Reply:
That's with a negative scale. It reminds me of double sidedness where you can mirror or not mirror and I couldn't find a common solution (Unity HDRP lets you choose which behavior you want).

Reply:
This has the effect of flipping T and N inside `main.vs`. 

B is still N x T (so handedness is the same) because we compute it in `shading_parameters.fs`.

ThreeJS renders the mirrored BusterDrone correctly without doing this explicitly because they use a traditional inverse-transpose rather than a cofactor matrix, and therefore already have the determinant sign accounted for.